### PR TITLE
Create plugin: Introduce sign-plugin

### LINKED ISSUE #115: Add yarn sign step into generated plugins
Currently generated scripts have no signing command, our CI relies on this command being there 
- https://github.com/grafana/plugin-tools/blob/main/packages/create-plugin/templates/github/ci/.github/workflows/release.yml#L80 

We should adjust the generated command to include sign and to utilize the new sign package

### PR DESCRIPTION
**Pull request by jackw**
**What this PR does / why we need it**:
The release workflow that is scaffolded by create-plugin has a dependency on toolkit to sign plugins. Now sign-plugin is available we can leverage it instead. This PR does the following:

- [x] updates workflows to use more modern approach to node module caching
- [x] add npm sign script (using `@grafana/sign-plugin`) to package.json
- [x] add a custom plop action to print a message when a plugin scaffold task is done
- [x] adds some docs related to distributing plugins / signing plugins to plugin README
- [x] updates release workflow to publish a draft github release with instructions for submission to grafana and auto generated release notes...

Screenshot 2022-11-24 at 17 40 58

**Which issue(s) this PR fixes**:

Fixes #115

**Special notes for your reviewer**:

<details>
  <summary>📦 Published PR as canary version: <code>Canary Versions</code></summary>
  <br />

  :sparkles: Test out this PR locally via:
  
  ```bash
  npm install @grafana/create-plugin@0.7.0-canary.160.9695113.0
  # or 
  yarn add @grafana/create-plugin@0.7.0-canary.160.9695113.0
  ```
</details>

### CODE REVIEW

**Review by leventebalogh — APPROVED** [review]
Amazing work mate! 👏

File: packages/create-plugin/src/commands/generate.plopfile.ts [hunk 1]
```diff
@@ -221,3 +231,27 @@ function getExportPath(pluginName: string, orgName: string, pluginType: PLUGIN_T
...
+
+function printSuccessMessage(answers: CliArgs) {
+  const directory = normalizeId(answers.pluginName, answers.orgName, answers.pluginType);
+
+  const commands = [
+    '  * `yarn install` to install frontend dependencies.',
+    '  * `docker-compose up` to start a grafana development server.',
+    '  * `yarn dev` to build (and watch) the plugin frontend code.',
+    ...(answers.hasBackend ? ['  * `mage -v build:linux` to build the plugin backend code.'] : []),
+    '  * Open http://localhost:3000/dashboard/new in your browser to create a dashboard to begin developing your plugin.',
```
[the remark this answers is not in the document]
Comment (jackw) [reply packages/create-plugin/src/commands/generate.plopfile.ts]: Thanks for spotting this. I did all the changes in a scaffolded panel plugin to make it easier to test but you're right we should land them at the root and let them do their thing. 😄

File: packages/create-plugin/templates/github/ci/.github/workflows/release.yml [hunk 2]
```diff
@@ -76,10 +53,15 @@ jobs:
           version: latest
           args: buildAll
 
+      - name: Warn missing Grafana API key
+        run: |
+          echo Please generate a Grafana API key: https://grafana.com/docs/grafana/latest/developers/plugins/sign-a-plugin/#generate-an-api-key
+          echo Once done please go to \"settings \> secrets \> actions\" to add it as GRAFANA_API_KEY
+        if: $\{{ env.GRAFANA_API_KEY == '' }}
```
[the remark this answers is not in the document]
Comment (jackw) [reply packages/create-plugin/templates/github/ci/.github/workflows/release.yml:60]: Totally agree. Tis updated! 👍

**Review by tolzhabayev — APPROVED** [review]
👏 nice work, we need to make sure to keep this doc about distributing your plugin and the one on our website in sync in the future but I think it's helpful to have this information close to the developer for a better DX

**Pull request merged**